### DISCUSSION TITLE
Journal backups needed

### DISCUSSION THREAD

**Discussion by Trebor87** [TechbeeAT/jtxBoard#862] [General]
Hi all

Firstly, before I forget. Bravo for JTX application which I find extremely good,  even though for the moment I am not really using it, but would love to

Let me explain if I may and please note these are not critics just my opinions

I really am interested in the Journal side of the application, For notes I have been using Joplin for years now and TBH works very well for me syncing automatically with my Drop Box account and far faster than syncing with NextCloud

For my daily Journal I use Diaro a online account and have been for almost ten years now, so I have many entries  as I write every day and add photos most days

Now JTX Journal is perfect for what I want, as good as Diaro and having the advantage to not being 'online' The only problem that I see is that it's 'local' with no sync / backup options

Ok I tried Davx5 for me, it's not useful -yet-  (sorry) . Two reasons, firstly images, it really doesn't like having images in your journal files, even if they are what I consider a normal size IE : 2 GB or more for a screen shot from my phone. 

Also even if I sync to my NextCloud account, nothing is seen in my Calendar., I believe that this due to NC not dealing with 'vjournal' . Also errors 504 are rather common, pretty sure that's the images screwing things up and making the sync too slow, (if I do my own NC account on my computer the 504 go away, But I'm shying away from NC as it's over complicated to set up in Linux and I don't really need it)

Davx5 for me needs to be able just to sync with something like DB, NC or GD just to make a backup

OR can't we have a 'standard' download file that includes the images and that of course could be re-uploaded if need be ? syncing to Drop Box, Google Drive for instance or even on the phone where the backup could be backup away from the phone. It doesn't need to be automatised, this the user's option to backup where and when he feels like it

Nearly all online Journals apps give that option in Text, Docx, for exemple, either to your PC or DB or GD for choice

Again these are not critics , I see the only decent Journal app around on Android that is not linked to a online account but for me sadly cannot be exploited as I believe it could be. Hope all this makes sense, I've been out of England for 45 years now and tend to forget my English

Best regards
Trevor

**Comment by patrickunterwegs** [comment]
Hey @Trebor87 
Thank you for your feedback!
Do I get it right that your description is mainly about the attachments?
Because basically without attachments the sync should just work fine with any CalDAV server (that supports the components).

About attachments: 
When I implemented jtx Board, I didn't want to spare the functionality despite I knew that servers would make troubles. Those iCalendar files are not meant to contain other files, so basically any attachment is stored as a base64 encoded string within an iCalendar file and most CalDAV servers have restrictions on that file size. So I decided to just make a warning for the users that this might cause problems. The limit is still up to the servers though.

A solution for this problem are managed attachments. This would mean that once an entry is synchronised to the server, the attachment is saved as a separate file on the server and the reference for the attachment replaced by a link to the attachment on the server. This then needs to be handled by DAVx5 and jtx Board.
It is an interesting thing, it just needs a lot of attention and development effort, los on DAVx5 side. So it will just take some time and there are still other priorities on DAVx5. But still, the topic is on the roadmap!

**Reply by Trebor87** [reply]
Hi

Thanks for getting back to me so quickly, yes I believe the problems that I've notices are due to attachments

As I don't use Calendars, or vary rarely, probably only for birthdays, (my wife would say use your memory not a calendar, She never uses one at all....) and I don't need my journal history in the calendar, also if and when I use one, it's Google and I cant get DAVx5 to work with that, I use 2FA oauth protection just about everywhere and even if I generate a specif app code by Google it won't accept making a Journal, that's Google for you :(  so again calendar is a no no for me

This is why I see a download option, being useful, and or a the 'Local' file be saved to a sync / stockage platform, NC, Drop Box & Gdrive (in my case)

JTX reads the local file, space then is not a problem I sometimes have 4 maybe 6 images per daily input, which can easily amount to 10 / 20 mb of images

Anyway, keep up the good work, I'll keep the app, but again sadly it cannot be used in my case

**Comment by rogercreagh** [comment]
Clearly managed attachments are the best solution, but in the meantime other local (jtx and or individual use-case specific) options are possible.

One for jtxB would be to enable markdown image syntax which can refer to a remote file and allows the picture to be embedded in the text with no server overhead. The application (jtx) can choose to cache the remote image if it wishes, or simply load it from the remote every time - which might not be much of a delay if the remote image is either on the same LAN or localhost (which would be ok if you dont mind images being unavailable when away from the LAN (unless jtxB had an option to cache remote images...)

`![alttext](https://server.dom/filename)` in markdown in jtxB causes it to crash out - which is annoying as it should ignore unrecognised markdown and display it as text - its the `!` it doesn't like. Can this bug be fixed please?

For the user my solution at present is to save images as a link to a remote server, rather than embed them in the ics. This gets around the server/davx5 timeout problems, but does mean jtxB fires up the browser to view them. My photos are all on a server anyway, so having to upload is not a problem.

If (or when) I was more organised I'd create a gallery on the server for each journal entry with multiple images so I could see them all at once. Not as good as embedding in markdown text, but it works.

If you only ever want to see the images on the phone/tablet you wrote jpurnal entries on, then installing a mini webserver like Simple Http Server app which lets you define any folder (eg your pictures or photos folder) as the localhost root and then you can link the images as http://127.0.0.1/filename

Which would be a brilliant solution if you could use markdown image embedding....

**Reply by Trebor87** [reply]
Hi

I tried your option
"For the user my solution at present is to save images as a link to a remote server... "

Using a image link from my Gdrive account,  this  works, but as you say it  ' fires' up the browser'  to see the image, or using a link from Garmin Connect also works, but it is a bit longer and  fiddlier to do than just adding a image. 
I can also link a photo directly from the phone photo app, again fiddlier, you need to copy paste the link,  of course, delete the photo on the phone and the link no longer works

I just quickly modified the couple of journal inputs that I wrote, changing the images for links, but still got a 504 error and it's still syncs longer than if I just have  'Local' inputs, but that's probably due to NextCloud, I've always found NC to be slow for syncing, but haven't found anything to replace it

I don't use markdown, never had a need, my journal logs are like this page, just info, that I can re-read in the future.
